### PR TITLE
refactor!: Make the driver fully compatible with Appium 2

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
BREAKING CHANGES:
- The driver is not compatible with Appium 1 anymore

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]

File: test/functional/caps-e2e-specs.js [hunk 1]
```diff
@@ -34,13 +34,16 @@ describe('Mac2Driver - caps', function () {
     }
   });
   beforeEach(async function () {
-    driver = wd.promiseChainRemote(HOST, PORT);
-    await driver.init(CAPS);
+    driver = await remote({
+      hostname: HOST,
+      port: PORT,
```
Comment (boneskull) [inline test/functional/caps-e2e-specs.js:39]: using a static port _may_ be problematic; use [get-port](https://npm.im/get-pport) if possible.

File: test/functional/find-e2e-specs.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ chai.use(chaiAsPromised);
 
 const CAPS = {
   platformName: 'mac',
-  bundleId: TEXT_EDIT_BUNDLE_ID,
+  'appium:bundleId': TEXT_EDIT_BUNDLE_ID,
```
Comment (boneskull) [inline test/functional/find-e2e-specs.js:13]: what's this?

**Review by boneskull — APPROVED** [review]
LGTM

I wonder how many drivers we need to convert to use wdio... is it worth writing a codemod?

File: test/functional/find-e2e-specs.js (hunk 2 above)
Comment (mykola-mokhnach) [reply test/functional/find-e2e-specs.js:13]: this is a valid capability name for this driver. Check https://github.com/appium/appium-mac2-driver#capabilities

File: test/functional/caps-e2e-specs.js (hunk 1 above)
Comment (mykola-mokhnach) [reply test/functional/caps-e2e-specs.js:39]: is there any useful example I can use?

**Comment by mykola-mokhnach** [comment]
> I wonder how many drivers we need to convert to use wdio... is it worth writing a codemod?

Basically all of them:
- xcuitest
- android
- uiautomator2
- espresso
- gecko
- safari
- windows

File: test/functional/caps-e2e-specs.js (hunk 1 above)
Comment (jlipps) [reply test/functional/caps-e2e-specs.js:39]: I think I merged a PR a while back that incorporated it into a test suite. Maybe a plugin? It seems like we'll want it to be a test convenience helper

**Pull request merged**